dags/common/parsing/xml_extractors.py

```diff
@@ -1,10 +1,68 @@
 import xml.etree.ElementTree as ET
 
+from common.constants import WHITE_SPACES
 from common.parsing.extractor import IExtractor
 from common.utils import check_value
 from structlog import get_logger
 
 
+class ContentExtractor(IExtractor):
+    def __init__(
+        self,
+        destination,
+        source,
+        required=True,
+        default_value=None,
+        extra_function=lambda s: s,
+        prefixes=None,
+    ) -> None:
+        super().__init__(destination)
+        self.destination = destination
+        self.source = source
+        self.prefixes = prefixes
+        self.required = required
+        self.default_value = default_value
+        self.extra_function = extra_function
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_extracted_value(self, node):
+        try:
+            values_in_regular_tags = [node.text or ""]
+            values_math_expression_and_styling_tags = [
+                ET.tostring(el).decode("ascii") for el in node
+            ]
+            extracted_value = "".join(
+                values_in_regular_tags + values_math_expression_and_styling_tags
+            )
+            return WHITE_SPACES.sub("", extracted_value)
+        except AttributeError:
+            self.logger.error(f"{self.destination} is not found in XML")
+            return
+
+    def _process_text_with_extra_function(self, text):
```

Comment:
_process_text_with_extra_function, extract ar the same in ContentExtractor and in TextExtractor. Should I move these functions to a common file and reuse it here? O better to leave, that code would be more easily readable?

Reply:
hm maybe you can move it to `IExtractor`? It won't be a super-pure interface (do I assume correctly that I come from an interface?), but well, firstly, it's not Java. Or maybe if you don't want to break Java good practices do a class that will implement this interface and add there this method.

Reply:
Removed the Content extractor, extraction function added to TextExtractr. Now, when we want to extract everything between tags, we will need to pass flag in text_extractor:
`all_content_between_tags=True`